### PR TITLE
Gatsby 2.0

### PR DESCRIPTION
**Pull request by ckj**
This PR:

- [Migrates to Gatsby 2.0](https://next.gatsbyjs.org/docs/migrating-from-v1-to-v2/)
- Fixes an issue where `firebase.auth()` would cause the build to fail
- Formats the code via `npm run format`

### CODE REVIEW

**Comment by rwieruch** [comment]
Oh wow, this is awesome! 

If I remember correctly, I introduced gatsby-next because of this issue with React's context API: https://github.com/taming-the-state-in-react/gatsby-firebase-authentication/issues/1 I guess this stays fixed then. Did you experience any bugs there? Otherwise I merge this. Thanks for your efforts! Helps a lot! 💯

**Comment by ckj** [comment]
> If I remember correctly, I introduced gatsby-next because of this issue with React's context API: #1 I guess this stays fixed then. 

I was told in [this issue](https://github.com/gatsbyjs/gatsby/issues/6668#issuecomment-407455771) that it was no longer necessary in v2. I personally didn't experience any issues but it might something worth confirming on your end before merging.

Happy to help!

**Comment by rwieruch** [comment]
Thanks again!

File: package.json [hunk 1]
```diff
@@ -4,11 +4,12 @@
   "version": "1.0.0",
   "author": "Robin Wieruch <wrobin@gmx.net>",
   "dependencies": {
+    "bluebird": "^3.5.1",
```
Comment (rwieruch) [inline package.json:7]: @ckj do you know where Bluebird is used in the application?

**Pull request merged**